src/message.ts

```diff
@@ -414,6 +414,93 @@ export class Message implements Sayable {
     return fromId === userId
   }
 
+  /**
+   *
+   * Get the mentioned contact which the message is mentioned for.
+   * message event table
+   *
+   * |                                                                            | Web  |  Mac PC Client | iOS Mobile |  android Mobile |
+   * | :---                                                                       | :--: |     :----:     |   :---:    |     :---:       |
+   * | [You were mentioned] tip ([有人@我]的提示)                                   |  ✘   |        √       |     √      |       √         |
+   * | Identify magic code (8197) by copy & paste in mobile                       |  ✘   |        √       |     √      |       ✘         |
+   * | Identify magic code (8197) by programming                                  |  ✘   |        ✘       |     ✘      |       ✘         |
+   * | Identify two contacts with the same roomAlias by [You were  mentioned] tip |  ✘   |        ✘       |     √      |       √         |
+   *
+   * @returns {Contact[]} return the contactList which the message is mentioned for
+   *
+   * @example
+   * ```ts
+   * const content = message.content()
+   * const contactList = message.mentioned()
+   * console.log(content)
+   * console.log(contactList)
+   * ```
+   */
+  public mentioned(): Contact[] {
+    let contactList: Contact[] = []
+    const room = this.room()
+    if (this.type() !== MsgType.TEXT || !room ) {
+      return contactList
+    }
+
+    // Example: `Let me introduce @lijiarui@wechaty@beijing , she is a wechaty contributor`
+    // Trying to split array with magic code `8197` and get atStrings
+    // Example result: atStrings = ['Let me introduce @lijiarui@wechaty@beijing', ', she is a wechaty contributor']
+    const atStrings = this.content().split(String.fromCharCode(8197))
+
+    if (atStrings.length === 0) return contactList
+
+    // Using `filter(e => e.indexOf('@') > -1)` to filter the string without `@`
+    // Example result: atStrings.filter(e => e.indexOf('@') > -1) = ['Let me introduce @lijiarui@wechaty@beijing']
+    const strings = atStrings.filter(e => e.indexOf('@') > -1).map(e => {
```

Comment:
1. Do not use var name `strings` because it will be very confusing with `string`
1. Do not use a map inline function with so many lines.
1. Do not use `e` as the var name. Because `e` stands for `event`, `error`, etc. use `s`/`str` for the string instead.

I can not understand the code inside this map function. 

However, I would like to rewrite this huge map function like this:

```ts
const mentionedStringList = atStrList
  .filter(str => str.include('@))
  .map(str => str.replace(/^.*?@/, '@'))
```